app/lib/encryption/encryption.js

```diff
@@ -0,0 +1,420 @@
+import EJSON from 'ejson';
+import SimpleCrypto from 'react-native-simple-crypto';
+import { Q } from '@nozbe/watermelondb';
+
+import {
+	toString,
+	utf8ToBuffer,
+	splitVectorData,
+	joinVectorData,
+	randomPassword
+} from './utils';
+import {
+	E2E_PUBLIC_KEY,
+	E2E_PRIVATE_KEY,
+	E2E_RANDOM_PASSWORD_KEY,
+	E2E_STATUS,
+	E2E_MESSAGE_TYPE,
+	E2E_ROOM_TYPES
+} from './constants';
+import RocketChat from '../rocketchat';
+import EncryptionRoom from './encryption.room';
+import UserPreferences from '../userPreferences';
+import database from '../database';
+import protectedFunction from '../methods/helpers/protectedFunction';
+
+class Encryption {
+	constructor() {
+		this.ready = false;
+		this.privateKey = null;
+		this.roomInstances = {};
+	}
+
+	// Initialize Encryption client
+	initialize = () => {
```

Comment:
It's to prevent some memory leak or improve performance?
IMO doing the initialization of all `collections.get` on `initialize` and starting to use class properties can be a mess to readability. WDYT?